### ISSUE TITLE
No address associated with hostname

### ISSUE THREAD

**Issue by jaimelaborda** [berkutta/parts_manager#16]
Hi,

I am trying to run parts_manager as docker cointainer but i could not get it working. I'm following all the steps you describe in the README.md but I am getting this error:

>    Illuminate\Database\QueryException  : SQLSTATE[HY000] [2002] php_network_getaddresses: getaddrinfo failed: No address associated with hostname (SQL: select * from information_schema.tables where table_schema = db and table_name = migrations and table_type = 'BASE TABLE')
> 
>   at /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php:669
>     665|         // If an exception occurs when attempting to run a query, we'll format the error
>     666|         // message to include the bindings with SQL, which will make this exception a
>     667|         // lot more helpful to the developer instead of just the database's errors.
>     668|         catch (Exception $e) {
>   > 669|             throw new QueryException(
>     670|                 $query, $this->prepareBindings($bindings), $e
>     671|             );
>     672|         }
>     673|
> 
>   Exception trace:
> 
>   1   Doctrine\DBAL\Driver\PDOException::("SQLSTATE[HY000] [2002] php_network_getaddresses: getaddrinfo failed: No address associated with hostname")
>       /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:31
> 
>   2   PDOException::("PDO::__construct(): php_network_getaddresses: getaddrinfo failed: No address associated with hostname")
>       /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:27
> 
>   Please use the argument -v to see more details.
> 
>    Illuminate\Database\QueryException  : SQLSTATE[HY000] [2002] php_network_getaddresses: getaddrinfo failed: No address associated with hostname (SQL: select * from `components` order by `id` asc limit 500 offset 0)
> 
>   at /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php:669
>     665|         // If an exception occurs when attempting to run a query, we'll format the error
>     666|         // message to include the bindings with SQL, which will make this exception a
>     667|         // lot more helpful to the developer instead of just the database's errors.
>     668|         catch (Exception $e) {
>   > 669|             throw new QueryException(
>     670|                 $query, $this->prepareBindings($bindings), $e
>     671|             );
>     672|         }
>     673|
> 
>   Exception trace:
> 
>   1   Doctrine\DBAL\Driver\PDOException::("SQLSTATE[HY000] [2002] php_network_getaddresses: getaddrinfo failed: No address associated with hostname")
>       /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:31
> 
>   2   PDOException::("PDO::__construct(): php_network_getaddresses: getaddrinfo failed: No address associated with hostname")
>       /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:27
> 
>   Please use the argument -v to see more details.
> 

Could you plase help me? It seems to be related with the database, What are the defaults user/pass you have used?

Thank you very much.

Best regards,
Jaime.

**Comment by berkutta** [comment]
Hi,

Please check your MySQL setup. Did you setup a MySQL Database and put valid MySQL credentials into the .env file?

**Comment by jaimelaborda** [comment]
Hi berkutta,

Do I need to setup a MySQL server instance or it is already setup inside the docker container? For me it is not clear.

Thank you.

Cheers,
Jaime.

**Comment by berkutta** [comment]
You need to provide a separate MySQL Server.

**Comment by jaimelaborda** [comment]
Hi,
Thank you for your reply.

No manner I can have this running. What mysql version are you using? I can't manage to properly connect to the database. Are you using OLD_PASSWORD format?

Best regards,
J.

**Comment by berkutta** [comment]
Hi,

Yes, it doesn't support the default auth in MySQL 8. You can change it with following SQL Query:
```
ALTER USER 'user'@'%' IDENTIFIED WITH mysql_native_password BY '1234';
```

**Comment by berkutta** [comment]
Hi,

Were you able to get it working? (Saw that you did a fork with a docker compose).

**Comment by jaimelaborda** [comment]
Hi, 

Sorry I did not have time to come back to you. I have been able to get it working with mysql version 5.7 and as you have seen I have forked the repo and added a docker compose file to set up a mysql instance and connect to parts_manager container. It is working quite well although parts_manager container starts before mysql and I have to manually restart it to catch the mysql server correctly (I know it can be solve with docker compose, I will try to do it when i have enought spare time).

Thank you for your help, I love this project! I would love to contribute but unfurtunatelly I don't have much experience with php and laravel framework :(

Cheers,
Jaime.
